org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/commands/CommandBuilder.java

```diff
@@ -0,0 +1,202 @@
+package org.metaborg.spoofax.shell.commands;
+
+import org.metaborg.core.MetaborgException;
+import org.metaborg.core.action.ITransformAction;
+import org.metaborg.core.language.ILanguageImpl;
+import org.metaborg.core.project.IProject;
+import org.metaborg.spoofax.shell.client.IResult;
+import org.metaborg.spoofax.shell.functions.FailableFunction;
+import org.metaborg.spoofax.shell.functions.IFunctionFactory;
+import org.metaborg.spoofax.shell.output.AnalyzeResult;
+import org.metaborg.spoofax.shell.output.EvaluateResult;
+import org.metaborg.spoofax.shell.output.InputResult;
+import org.metaborg.spoofax.shell.output.ParseResult;
+import org.metaborg.spoofax.shell.output.TransformResult;
+
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+
+/**
+ * Builds commands by composing several smaller functional interfaces.
```

Comment:
Which are? You can (and should, imo) put more effort in the JavaDoc. Our work is going to be picked up by others, so having good documentation is a must.